### PR TITLE
feat: PHPStan Request Factories - Upgrade to PHPStan 2.x

### PR DESCRIPTION
**Pull request by odinns**
Upgraded to PHPStan 2.x for a 2.0 release.

- [x] Waiting for Rector 2.x to be released.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,15 +19,15 @@
         "php": "^8.2",
         "illuminate/contracts": "^10.0 || ^11.0",
         "illuminate/http": "^10.0 || ^11.0",
-        "phpstan/phpstan": "^1.10.57",
+        "larastan/larastan": "^3.0",
```
Comment (owenvoke) [inline composer.json]: I don't think Larastan is needed as a non-dev requirement here? Same with the Coding Style. They should probably be moved back to `require-dev` 👍🏻

Comment (odinns) [reply composer.json]: Alright 👍

**Review by owenvoke — APPROVED** [review]
Fine with me. 👍🏻 Happy for this to be merged now, or can wait until Rector 2.x is released for https://github.com/worksome/coding-style/pull/80

**Comment by odinns** [comment]
> Fine with me. 👍🏻 Happy for this to be merged now, or can wait until Rector 2.x is released for [worksome/coding-style#80](https://github.com/worksome/coding-style/pull/80)

I will wait until Rector 2.x is released so I don't have to make a new PR to change the coding-style dependency.

We can reference this branch now when testing upgrading platform to PHPStan 2 😊

**Review by owenvoke — APPROVED** [review]

**Comment by owenvoke** [comment]
Just pushed to update to Coding Style 3.x, so this is ready to go. 👍🏻

**Pull request merged**